src/components/space-panel/space-panel.tsx

```diff
@@ -80,18 +80,20 @@ export const SpacePanel = ({
       {isSignedIn
         ? <SpacePanelIframe cloudBaseURL={cloudBaseURL} streamedHostsData={streamedHostsData} />
         : (
-          <S.ScrollContainer>
-            {hasStreamedHosts && (
-              <S.PanelSection leading>
-                <ReplicatedNodes chartsMetadata={chartsMetadata} />
-              </S.PanelSection>
-            )}
-            {(machinesArray.length > 0) && (
-              <S.PanelSection leading={!hasStreamedHosts}>
-                <VisitedNodes machinesArray={machinesArray} />
-              </S.PanelSection>
-            )}
+          <>
+            <S.ScrollContainer>
+              {hasStreamedHosts && (
+                <S.PanelSection leading>
```

Comment:
Maybe we could create component with a prop `isVisible` to use react optimizations?

Reply:
That sounds interesting, can you briefly say what it is, or paste a link? You mean to render the component always, and if it's not needed, then it should return `null`, and it is then faster?

If yes, it's interesting idea and there would be places where it could come handy (ie. on chart level for example), but here we only render it once so it doesn't make that much difference

Reply:
> ou mean to render the component always, and if it's not needed, then it should return null, and it is then faster?


@jacekkolasa Yes, thats was the idea